pkg/csi/service/common/vsphereutil.go

```diff
@@ -178,8 +175,98 @@ func CreateBlockVolumeUtil(ctx context.Context, clusterFlavor cnstypes.CnsCluste
 	return volumeID.Id, nil
 }
 
-// CreateFileVolumeUtil is the helper function to create CNS file volume.
-func CreateFileVolumeUtil(ctx context.Context, clusterFlavor cnstypes.CnsClusterFlavor, manager *Manager, spec *CreateVolumeSpec) (string, error) {
+// CreateFileVolumeUtil is the helper function to create CNS file volume with datastores.
+func CreateFileVolumeUtil(ctx context.Context, clusterFlavor cnstypes.CnsClusterFlavor,
+	manager *Manager, spec *CreateVolumeSpec, datastores []*vsphere.DatastoreInfo) (string, error) {
+	log := logger.GetLogger(ctx)
+	vc, err := GetVCenter(ctx, manager)
+	if err != nil {
+		log.Errorf("failed to get vCenter from Manager, err: %+v", err)
+		return "", err
+	}
+	if spec.ScParams.StoragePolicyName != "" {
+		// Get Storage Policy ID from Storage Policy Name
+		spec.StoragePolicyID, err = vc.GetStoragePolicyIDByName(ctx, spec.ScParams.StoragePolicyName)
+		if err != nil {
+			log.Errorf("Error occurred while getting Profile Id from Profile Name: %q, err: %+v", spec.ScParams.StoragePolicyName, err)
+			return "", err
+		}
+	}
+	var datastoreMorefs []vim25types.ManagedObjectReference
+	if spec.ScParams.DatastoreURL == "" {
+		datastoreMorefs = getDatastoreMoRefs(datastores)
+	} else {
+		// If datastoreUrl is set in storage class, then check if part of input datastores.
+		// If true, create the file volume on the datastoreUrl set in storage class.
+		var isFound bool
+		for _, dsInfo := range datastores {
+			if spec.ScParams.DatastoreURL == dsInfo.Info.Url {
+				isFound = true
+				datastoreMorefs = append(datastoreMorefs, dsInfo.Reference())
+				break
+			}
+		}
+		if !isFound {
+			msg := fmt.Sprintf("CSI user doesn't have permission on the datastore: %s specified in storage class",
+				spec.ScParams.DatastoreURL)
+			log.Error(msg)
+			return "", errors.New(msg)
+		}
+	}
+
+	// Retrieve net permissions from CnsConfig of manager and convert to required format
+	netPerms := make([]vsanfstypes.VsanFileShareNetPermission, 0)
+	for _, netPerm := range manager.CnsConfig.NetPermissions {
```

Comment:
What would be the net permission applied in case of WCP when the file volume is created?